src/bun.js/javascript.zig

```diff
@@ -1318,6 +1319,12 @@ pub const VirtualMachine = struct {
     }
 
     pub fn onBeforeExit(this: *VirtualMachine) void {
+        Bun__queueFinishNapiFinalizers(this.global);
+        while (this.isEventLoopAlive()) {
```

Comment:
This should only be called if there were napi finalizers, right?

Reply:
I changed it to execute the while loop only if there are any pending finalizers.

Reply:
Actually, that causes test failures in test/js/node/process/process.test.js. It seems we have to tick the event loop a bit regardless of whether the NAPI finalizer queue is empty. This is probably because of the call to `vm.heap.addFinalizer` in `napi_add_finalizer`.

Reply:
Never mind. I [found](https://github.com/oven-sh/bun/commit/a8fa566101e3ec17eb6b9aea7fc00dab9ba2c15e) the underlying issue. Now it will tick the event loop only if necessary while also not dispatching onBeforeExit multiple times.